commit c5d34904ab95b85938361280e5540315eeb03d8c
Author: bertin <macportsuser@macports.org>
Date:   Thu May 22 23:41:26 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-truncate-logfile.diff

diff --git a/src/macports1.0/macports.tcl b/src/macports1.0/macports.tcl
index e65a09cb2..d0c0315ed 100644
--- a/src/macports1.0/macports.tcl
+++ b/src/macports1.0/macports.tcl
@@ -289,8 +289,8 @@ proc macports::ch_logging {mport} {
     file mkdir $logdir
     set debuglogname [file join $logdir main.log]
 
-    # Append to the file if it already exists
-    set debuglog [open $debuglogname a]
+    # Truncate the file if it already exists
+    set debuglog [open $debuglogname w]
     puts $debuglog version:1
 
     ui_debug "Starting logging for $portname @[dict get $portinfo version]_[dict get $portinfo revision][dict get $portinfo canonical_active_variants]"
